kotlinx-coroutines-core/common/test/channels/ClosedChannelTest.kt

```diff
@@ -0,0 +1,196 @@
+package kotlinx.coroutines.channels
+
+import kotlinx.coroutines.testing.*
+import kotlin.test.*
+
+class ClosedChannelTest : TestBase() {
+    /**
+     * Iterator.
+     */
+
+    @Test
+    fun testIteratorClosedHasNextFalse() = runTest {
+        TestChannelKind.entries.forEach { kind ->
+            val channel = kind.create<Int>()
+            val iterator = channel.iterator()
+            channel.close()
+            assertFalse(iterator.hasNext())
+        }
+    }
+
+    @Test
+    fun testIteratorClosedWithExceptionHasNextThrows() = runTest {
+        TestChannelKind.entries.forEach { kind ->
+            val channel = kind.create<Int>()
+            val iterator = channel.iterator()
+            channel.close(TestException())
+            assertFailsWith<TestException> { (iterator.hasNext()) }
+        }
+    }
+
+    @Test
+    fun testClosedToListOk() = runTest {
+        TestChannelKind.entries.forEach { kind ->
+            val channel = kind.create<Int>()
+            channel.close()
+            channel.toList()
+        }
+    }
+
+    @Test
+    fun testClosedWithExceptionToListThrows() = runTest {
+        TestChannelKind.entries.forEach { kind ->
+            val channel = kind.create<Int>()
+            channel.close(TestException())
+            assertFailsWith<TestException> { channel.toList() }
+        }
+    }
+
+    @Test
+    fun testClosedConsumeEachOk() = runTest {
```

Comment:
Duplicates `ConsumeTest#testConsumeEachClosesOnSuccess`, and I'd argue that place is a better fit for this test, since `close` is the more fundamental operation.